pkg/eval/platform/platform.go

```diff
@@ -35,9 +37,46 @@ import (
 // Whether or not the platform is Microsoft Windows.
 // This is read-only.
 
+//elvdoc:fn hostname &omit-domain=false
+//
+// The host name of the system. If option `&omit-domain` is used any domain
+// components are stripped from the host name. If an error occurs while
+// fetching the host name `$nil` is substituted. This is implemented using
```

Comment:
It is better to propagate the error as an exception. If you let `hostname` return `(string, error)`, the error is automatically converted to an exception.

Reply:
Done, although I am ambivalent. It's not clear that this can ever fail in practice but if it does should that be fatal?